tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java

```diff
@@ -0,0 +1,239 @@
+
+            IStatus status = sdk.validate(true, false);
+
+            if ( !status.isOK() )
+            {
+                IStatus[] statuses = status.getChildren();
+
+                for( IStatus iStatus : statuses )
+                {
+                    ProjectUtil.setMarker(sdkProject, IMarker.PROBLEM, IMarker.SEVERITY_ERROR,
+                        iStatus.getMessage(), sdkProject.getFullPath().toPortableString(),ID_SDK_PROPERTIES_INVALID);
+                }
+
+                return;
+            }
+            sdk.getBuildProperties( true, true );
+        }
+
+        for( final IProject project : CoreUtil.getAllProjects() )
+        {
+            if ( !CoreUtil.isLiferayProject ( project ) )
+            {
+                continue;
+            }
+
+            new WorkspaceJob( "Updating project setting base on new sdk properties for " + project.getName() )
+            {
+                @Override
+                public IStatus runInWorkspace( IProgressMonitor monitor ) throws CoreException
+                {
+                    IJavaProject javaProject = JavaCore.create( project );
+
+                    IPath containerPath = null;
+
+                    IClasspathEntry[] entries = javaProject.getRawClasspath();
+
+                    for( IClasspathEntry entry : entries )
+                    {
+                        if( entry.getEntryKind() == IClasspathEntry.CPE_CONTAINER )
+                        {
+                            if( entry.getPath().segment( 0 ).equals( PluginClasspathContainerInitializer.ID ) ||
+                                            entry.getPath().segment( 0 ).equals( SDKClasspathContainer.ID ))
+                            {
+                                containerPath = entry.getPath();
+
+                                break;
+                            }
+                        }
+                    }
+
+                    if( containerPath != null )
+                    {
+                        IClasspathContainer classpathContainer = JavaCore.getClasspathContainer( containerPath, javaProject );
+
+                        final String id = containerPath.segment( 0 );
+
+                        if ( id.equals( PluginClasspathContainerInitializer.ID ) ||
+                             id.equals( SDKClasspathContainer.ID ) )
+                        {
+                            ClasspathContainerInitializer initializer = JavaCore.getClasspathContainerInitializer( id );
+                            initializer.requestClasspathContainerUpdate( containerPath, javaProject, classpathContainer );
+                        }
+                    }
+
+                    return Status.OK_STATUS;
+                }
+            }.schedule();
+        }
+    }
+
+    @Override
+    public void resourceChanged( IResourceChangeEvent event )
+    {
+        if( event == null )
+        {
+            return;
+        }
+
+        try
+        {
+            event.getDelta().accept( this );
```

Comment:
I think we need to handle add or change two type events. For user separately importing new plugin sdk project into workspace, we need to make sure theose projects based on the new imported plugin sdk should be updated by the new sdk classContainer.